Update README (#128)
Frogging-Family/linux-tkg: README.md

> ## Linux-TKG

Reviewer: Nit: Usually, isn't it "linux-tkg"?

Author: Haha I don't know, I thought since it's a title x) Lemme change that

Author: You suggest lowercase "L" too ?

Reviewer: Yeah. You're right that titles do use that casing, but the casing that the project is usually referred to with takes precedence.

Author: Done!

This repository provides scripts to automatically download, patch and compile the Linux Kernel from [the official Linux git repository](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git), with extra select patches that aim for better desktop/gaming performance. Users can customize which patches are added to the "vanilla" sources by editing the file `customization.cfg` and/or by following the interactive install script. Users can also provide their own patches: more information is available in the file `customization.cfg`.

### Important information

> - Due to `intel_pstate` poor performance as of late, it is set it to passive mode to make use of the `acpi_cpufreq` governor passthrough, keeping full support for turbo frequencies.

Reviewer: ```suggestion
- Due to `intel_pstate`'s poor performance as of late, it is set it to passive mode to make use of the `acpi_cpufreq` governor passthrough, keeping full support for turbo frequencies.
```

Author: Thanks!

### Important information

- In `intel_pstate` driver, frequency scaling aggressiveness has been changed with kernel 5.5 which results in stutters and poor performance in low/medium load scenarios (for higher power savings). As a workaround for our gaming needs, we are setting it to passive mode to make use of the `acpi_cpufreq` governor passthrough, keeping full support for turbo frequencies. It's combined with our aggressive ondemand governor by default for good performance on most CPUs while keeping frequency scaling for power savings. In a typical low/medium load scenario (Core i7 9700k, playing Mario Galaxy on Dolphin emulator) intel_pstate in performance mode gives a stuttery 45-50 fps experience, while passive mode + aggressive ondemand offers a locked 60 fps.
- Nvidia's proprietary drivers might need to be patched if they don't support your chosen kernel OOTB: [Frogging-Family nvidia-all](https://github.com/Frogging-Family/nvidia-all) can do that automatically for you.

> - Note regarding kernels older than 5.9 on `Archlinux`: since the switch to `zstd` compressed `initramfs` by default, you will face an `invalid magic at start of compress` error by default. You can workaround the issue by editing `/etc/mkinitcpio.conf` to uncomment the `COMPRESSION="lz4"` (for example, since that's the best option after zstd) line and regenerating `initramfs` for all kernels with `sudo mkinitpcio -P`

Reviewer: See [here](https://wiki.archlinux.org/title/Arch_terminology#Arch_Linux) for how to stylize Arch Linux.
```suggestion
- Note regarding kernels older than 5.9 on Arch Linux: since the switch to `zstd` compressed `initramfs` by default, you will face an `invalid magic at start of compress` error by default. You can workaround the issue by editing `/etc/mkinitcpio.conf` to uncomment the `COMPRESSION="lz4"` (for example, since that's the best option after zstd) line and regenerating `initramfs` for all kernels with `sudo mkinitpcio -P`
```

> - Note regarding kernels older than 5.9 on `Archlinux`: since the switch to `zstd` compressed `initramfs` by default, you will face an `invalid magic at start of compress` error by default. You can workaround the issue by editing `/etc/mkinitcpio.conf` to uncomment the `COMPRESSION="lz4"` (for example, since that's the best option after zstd) line and regenerating `initramfs` for all kernels with `sudo mkinitpcio -P`

Reviewer: ```suggestion
- Note regarding kernels older than 5.9 on `Archlinux`: since the switch to `zstd` compressed `initramfs` by default, you will face an `invalid magic at start of compress` error by default. You can workaround the issue by editing `/etc/mkinitcpio.conf` to uncomment the `COMPRESSION="lz4"` (for example, since that's the best option after zstd) line and regenerating `initramfs` for all kernels with `sudo mkinitpcio -P`.
```

### Customization options
#### Alternative CPU schedulers

> CFS is the only CPU scheduler available in the "vanilla" kernel sources. Its current implementation doesn't allow for injecting additional schedulers, and requires replacing it. Only one scheduler can be patched in at a time.

Reviewer: Since the AW links to this on its [own page on the subject](https://wiki.archlinux.org/title/Improving_performance#Alternative_CPU_schedulers).
```suggestion
[CFS](https://en.wikipedia.org/wiki/Completely_Fair_Scheduler) is the only CPU scheduler available in the "vanilla" kernel sources. Its current implementation doesn't allow for injecting additional schedulers, and requires replacing it. Only one scheduler can be patched in at a time.
```

> Alternative schedulers are available to you in tkg:

Reviewer: ```suggestion
Alternative schedulers are available to you in linux-tkg:
```

- [Graysky's per-CPU-arch native optimizations](https://github.com/graysky2/kernel_compiler_patch): tunes the compiled code to to a specified CPU
- Compile with `GCC` or `Clang`
  - Choice between `O2` and `O3` compiler optimizations.
  - `Clang` offers an experimental "Link Time Optimization" (`LTO`) feature that may improve the Kernel's performance (any potential improvement in gaming is yet to be proven). 
    - **Warning:** as of now, building any DKMS module will fail if this feature is used. Do not enable `LTO` if you need `DKMS` modules for proper kernel functioning (_e.g_ Nvidia's drivers). Otherwise the resulting kernel may be unbootable.
- Use [Modprobed-db](https://github.com/graysky2/modprobed-db)'s database to reduce the compilation time and produce a smaller kernel. The produced kernel will only contain the modules listed in that database.
  - **Warning**: make sure to read [thoroughly about it first]((https://wiki.archlinux.org/index.php/Modprobed-db)) since it comes with caveats making it **NOT recommended for most users**. Otherwise the resulting kernel may be unbootable.
- "Zenify" patchset using core blk, mm and scheduler tweaks from Zen

> - Anbox support (binder, ashmem)

Reviewer: ```suggestion
- [Anbox](https://wiki.archlinux.org/title/Anbox) support (via the `binder` and `ashmem` kernel modules)
```

- `ZFS` FPU symbols (<5.9)
- Overrides for missing ACS capabilities
- Provide own kernel `.config` file

> - ...

Reviewer: What's this?

Author: Three dots to say "and more..." x)
